### PR TITLE
Use Stopwatch for timing, not DateTime

### PR DESCRIPTION
**Pull request by mj1856**
Timing operations should always be done with a `System.Diagnostics.Stopwatch`.

Using `DateTime.Now` is undesirable because:

- The system clock is not very precise (RTC vs QPC).
- The system clock can drift.
- The system clock can be adjusted automatically via NTP sync, or manually by the user.
- `DateTime.Now` is based on the computer's local time zone, which may lead to errors if the timing operation crosses a daylight saving time transition.

### CODE REVIEW

**Comment by tsviatko** [comment]
Thanks for taking the time to diagnose this problem.

While the problem is there its severity is relatively low due to the fairly low occurrence frequency of DST transitions and the small corrections made by NTP. Moreover, there is a simple workaround, i.e. a second run.

Using the Stopwatch class also has [drawbacks](http://www.virtualdub.org/blog/pivot/entry.php?id=106) that might lead to imprecision. So further research is needed before picking up a better method.

To sum up, taking in mind the low impact of this bug and the time it will take fixing it both here and in JustDecompile's UI I'd rather spend the time on another issue with greater impact.

**Comment by mj1856** [comment]
Yes, I know that article well.  It is quite outdated though.  On modern platforms, QPC is much more reliable and is the recommended way to do this.  All of the issues discussed in that article have been resolved on modern hardware and OS platforms.  Read ["Dispelling the myths, rumors, and innuendo surrounding the QueryPerformanceCounter function"](http://blogs.msdn.com/b/oldnewthing/archive/2014/08/22/10551965.aspx) and ["Acquiring high-resolution time stamps"](https://msdn.microsoft.com/en-us/library/windows/desktop/dn553408.aspx).  These are also found in the comments at the bottom of the article you posted.

Also note that the article was posted on the blog of Virtualdub, a video editing software. For video processing, the recommended approach is to use [multimedia timers](https://msdn.microsoft.com/en-us/library/dd743609.aspx), which while offering higher precision, are much more difficult to use in managed code, and can have some system performance impacts, which is why `Stopwatch` implements QPC rather than MM timers. (MM timers would be overkill for your scenario.)

Low priority - fine.  But you guys openly asked for help finding bugs without any constraints or guidelines on what to focus on, so I sent you the first thing I saw.

**Comment by mj1856** [comment]
One last point - you'd be surprised how often the operating system adjusts the clock all by itself.  In testing, I've found most systems correct for several milliseconds of drift every 5 to 15 minutes.  I'll blog about it at some point, but believe me - the clock is not as reliable as you would think.

**Comment by tsviatko** [comment]
We would definitely like to hear about every bug our users encounter. Rather unfortunately, we simply don't have the resources to fix every bug we are aware of and that's why we have to do some cost-benefit analysis and prioritize.

Since our desired precision is somewhere in between 0.1 and 1 seconds DST would be the only real problem with our current time measuring approach. The impact of that does not justify spending more time on further research on the subject at present and it looks like we have to do more research to verify QPC is the best approach currently available. So, I am going to add this issue to our bug backlog and close the pull request.

Nonetheless, we really appreciate the time and effort you spent on helping us improve the product. Again we  are really happy to have our community help us find any bugs in our code. So, please, do not hesitate to submit fixes to other problems you encounter.

**Comment by mj1856** [comment]
At a minimum, you should consider using `DateTime.UtcNow`, which is not affected by DST transitions.   it still uses the same clock source.

**Comment by tsviatko** [comment]
Yes, we can easily do that and that change comes at a very low cost since no further research is needed. That'd solve the DST problem which is the primary source of concern with this issue.

Thanks a bunch for pointing out this elegant solution. The code will be changed accordingly.

I really appreciate you bearing with me this far on this issue. Once again, please, do not hesitate to submit fixes to other problems you encounter.
